lambda/matching.ml

```diff
@@ -260,15 +263,11 @@ module Simple : sig
   val head : pattern -> Patterns.Head.t
 
   val explode_or_pat :
-    Half_simple.pattern * Typedtree.pattern list ->
-    arg_id:Ident.t option ->
+    arg:lambda ->
+    Half_simple.pattern ->
     mk_action:(vars:Ident.t list -> lambda) ->
     vars:Ident.t list ->
```

Comment:
Coming back to this code after some months, I couldn't remember straightaway what `vars` was, reading its implementation I could venture a guess, which I confirmed by looking at the (sole) call-site of this function.

At the said call-site, there's the following comment just above `vars` which I think could be moved up here:
> (* bound variables of the or-pattern and used in the orpm actions *)

Reply:
We might want to try to wait 3 months on every PR to see if the code still makes sense :-) Thanks for the excellent nitpick.

Reply:
Rebased. I renamed `vars` into `patbound_action_vars`, is that clearer? I did not end up moving the comment around due to decision fatigue (on the function signature or implementation? document just this parameter or also the rest?).